examples/auth-hook.py

```diff
@@ -0,0 +1,33 @@
+# Example how to roll your own Authorization in the dredd hooks
+import json
+import dredd_hooks as hooks
+import requests
+
+stash = {}
+api_username = "XXX"
+api_password = "XXX"
+api_uri = "XXX"
+
+
+@hooks.before_each
+def add_token(transaction):
+    if 'token' in stash:
+        print('adding a token')
+    else:
+        stash['token'] = obtain_token()
+
+    transaction['request']['headers']['Authorization'] = "Bearer " + stash['token']
+
+
+def obtain_token():
+    data = {
+               "api_login_form": {
+                   "username": api_username,
+                   "password": api_password
+               }
+           }
+    url = api_uri
+    response = requests.post(url, json=data)
```

Comment:
Might be a good idea to add a `response.raise_for_status()`. That way requests will throw an exception for non successful responses (400s 500s etc). These exceptions contain more context on server failures so it would be easier for a user to understand when there obtain token API is broken.